src/utils/nexmark/src/utils.rs

```diff
@@ -14,67 +14,49 @@
 
 use std::collections::HashMap;
 
-use chrono::NaiveDateTime;
 use rand::rngs::SmallRng;
 use rand::{Rng, SeedableRng};
 
 const MIN_STRING_LENGTH: usize = 3;
 
 pub trait NexmarkRng {
     fn gen_string(&mut self, max: usize) -> String;
-    fn gen_string_with_delimiter(&mut self, max: usize, delimiter: &str) -> String;
+    fn gen_string_with_delimiter(&mut self, max: usize, delimiter: char) -> String;
     fn gen_exact_string(&mut self, length: usize) -> String;
     fn gen_next_extra(&mut self, current_size: usize, desired_average_size: usize) -> String;
     fn gen_price(&mut self) -> usize;
 }
 
 impl NexmarkRng for SmallRng {
     fn gen_string(&mut self, max: usize) -> String {
-        self.gen_string_with_delimiter(max, " ")
+        self.gen_exact_string(max)
     }
 
-    fn gen_string_with_delimiter(&mut self, max: usize, delimiter: &str) -> String {
+    fn gen_string_with_delimiter(&mut self, max: usize, delimiter: char) -> String {
         let len = self.gen_range(MIN_STRING_LENGTH..max);
-        String::from(
-            (0..len)
-                .map(|_| {
-                    if self.gen_range(0..13) == 0 {
-                        delimiter.to_string()
-                    } else {
-                        ::std::char::from_u32('a' as u32 + self.gen_range(0..26))
-                            .unwrap()
-                            .to_string()
-                    }
-                })
-                .collect::<Vec<String>>()
-                .join("")
-                .trim(),
-        )
+        (0..len)
+            .map(|_| {
+                if self.gen_range(0..13) == 0 {
+                    delimiter
+                } else {
+                    self.gen_range(b'a'..=b'z') as char
+                }
+            })
+            .collect()
     }
 
     fn gen_exact_string(&mut self, length: usize) -> String {
-        let mut s = String::new();
-        let mut rnd = 0;
-        let mut n = 0;
-        for _ in 0..length {
-            if n == 0 {
-                rnd = self.gen();
-                n = 6; // log_26(2^31)
-            }
-            let c = ::std::char::from_u32('a' as u32 + (rnd % 26)).unwrap();
-            s.push(c);
-            rnd /= 26;
-            n -= 1;
-        }
-        s
+        (0..length)
+            .map(|_| self.gen_range(b'a'..=b'z') as char)
+            .collect()
     }
 
     fn gen_next_extra(&mut self, current_size: usize, desired_average_size: usize) -> String {
         if current_size > desired_average_size {
             return String::new();
         }
         let desired_average_size = desired_average_size - current_size;
-        let delta = (desired_average_size as f64 * 0.2).round() as usize;
+        let delta = desired_average_size / 5;
```

Comment:
Will this break the `round()` semantic?

Reply:
Sure.. Is that critical? We can fix it by:
```suggestion
        let delta = (desired_average_size + 2) / 5;
```